Assets/Scripts/Environment/Hangar/Battery.cs

```diff
@@ -55,38 +62,77 @@ private void SetEmission()
         meshRenderer.material.SetColor("_EmissionColor", emission);
     }
 
-    public bool PlayerIsHolding()
+    /*public bool PlayerIsHolding()
     {
-        return PlayerManager.PlayerObject.GetComponentInChildren<Battery>() != null;
+        return springJoint != null && springJoint.connectedBody == PlayerManager.PlayerMovement.PlayerRigidbody;
+
+        return PlayerManager.IsHoldingBattery;
     }
+    */
 
     public void TakeBattery()
     {
-        Container.ParentToPlayer();
+        Container.transform.localPosition = new Vector3(Container.transform.localPosition.x, HangarConstants.BatteryYPosition, Container.transform.localPosition.z);
+
+        containerRigidBody.useGravity = false;
+
+        // We add constraints so that the battery doesn't swing around uncontrollably
+        containerRigidBody.constraints = HangarConstants.BatteryRigidbodyConstraints;

```

Comment:
We might want to set the constraints on the prefab instead. At the moment, the rotation is not frozen when the battery is on the ground. So the battery can get in an odd position, and when the player picks it up again they can stand on top of it and float around.